nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -140,6 +143,26 @@ private void initNsisExecutable(GenericFactory factory)
 		return sections;
 	}
 
+	/**
+	 * Get an array of the right amount of entries in the Nsis executable. The
+	 * reader object is expected to be at the right offset (at the beginning of the
+	 * first entry) before calling this function. The reader index is advanced and
+	 * after executing this function, the index of the reader is pointing to the
+	 * first byte after the entries section.
+	 * 
+	 * @param reader
+	 * @return
+	 * @throws IOException
+	 */
+	private NsisEntry[] getEntries(BinaryReader reader) throws IOException {
+		NsisBlockHeader entriesBlockHeader = this.commonHeader.getBlockHeader(2);
```

Comment:
why getBlockHeader(2)? Where does 2 come from?

If 2 is the position of the entries block, please add named constants somewhere for block positions.